app/views/shared/_alert.html.erb

```diff
@@ -1,5 +1,14 @@
+<%#
+locals:
+* type: One of: "info", "success", "warning", "error", "other". Defaults to "info".
+* text_tag: Optional override HTML tag for text content. Defaults to `p`.
+* class: Additional class names to add to alert wrapper.
+* message: Text to display in alert. Text can also be provided through block.
+%>
 <%
-  type = local_assigns.fetch(:type, 'other')
+  type = local_assigns.fetch(:type, 'info')
+  type = 'error' if type === 'alert'
+  type = 'info' unless ['info', 'success', 'warning', 'error', 'other'].include?(type)
```

Comment:
Is `alert` a valid type? It's not listed in the comment. Also, is `error` supposed to get mapped to `info`?

Reply:
> Is `alert` a valid type? It's not listed in the comment.

The way I was considering it was that the valid types are only those listed here:

https://design.login.gov/components/alerts/

Unfortunately, we need to normalize to account for some other types, specifically those coming from third-party gems like Devise:

>Devise uses flash messages with I18n, in conjunction with the flash keys :notice and :alert.

https://github.com/heartcombo/devise#i18n

So I don't consider it "valid" as in what we'd want to encourage our own developers to be using, but need to account for it all the same.

There are a handful of instances in our code where we're using `:notice` and `:alert`. I think those should be updated. I have a `git stash` of these changes, but decided against including it here, for the sake of making review easier.

>Also, is `error` supposed to get mapped to `info`?

No, `error` wouldn't map to `info`. It's the inverse: Anything outside this array would map to `info`. In particular, we've got a handful of places where in code we just throw things into `flash` with arbitrary keys. Because these would appear with a blue background by default before, I chose to use `'info'` as the default appearance.

Reply:
>In particular, we've got a handful of places where in code we just throw things into flash with arbitrary keys. Because these would appear with a blue background by default before, I chose to use 'info' as the default appearance.

I think I misunderstood slightly how we're using flash messages, but not in a way which has much of an impact.

There's a hard-coded set of flash types we're expecting:

https://github.com/18F/identity-idp/blob/538cf31d37a741b0c63126b5ec30cdf104682b46/app/controllers/application_controller.rb#L10

So `_flashes.html.erb` won't actually render alerts aside from those types.

We may want to modify those based on the guidance above (remove `alert`, add `info` and `other`), but not necessarily a requirement here.

Reply:
Between 035316bbc, e5e9cb938, 76daf5e8f, 69df36977, the partial now only allows the expected set of types.